src/chai/ArrayManager.cpp

```diff
@@ -326,7 +327,7 @@ void ArrayManager::free(PointerRecord* pointer_record, ExecutionSpace spaceToFre
               if (space_ptr == pointer_record->m_pointers[space_t])
                 pointer_record->m_pointers[space_t] = nullptr;
             }
-          } else {
+          } else
```

Comment:
I think this `else` without an opening curly brace is problematic.

Reply:
I think so too - but also concerned that testing isn't catching a potential mismatch bracket issue?

Reply:
IMO, there should also be curly braces around line 328 above as well for readability, at least.

Reply:
So without the brace, was it just executing the callback line and missing the deallocate?

Reply:
Good catch.  Without the brace it is valid code, just not what I wanted, I think.  Only the first line will be in the conditional clause and the rest will be unconditional.

Reply:
without the brace, the deallocation would have been called twice in the other cases (other than the else).

Reply:
@liu15 thanks for the curly brace additions. I'm not as sad anymore :-)